code/modules/photography/camera/camera_image_capturing.dm

```diff
@@ -82,14 +78,23 @@
 	else
 		for(var/X in sorted) //these are clones
 			var/obj/effect/appearance_clone/clone = X
-			var/xo = (clone.x - center.x) * world.icon_size + clone.pixel_x + xcomp
-			var/yo = (clone.y - center.y) * world.icon_size + clone.pixel_y + ycomp
-			xo += clone.step_x
-			yo += clone.step_y
+			var/xo = (clone.x - center.x) * world.icon_size + clone.pixel_x + xcomp + clone.step_x
+			var/yo = (clone.y - center.y) * world.icon_size + clone.pixel_y + ycomp + clone.step_y
+			if(clone.transform)
+				xo -= world.icon_size * (clone.transform.get_x_scale() - 1) / 2
+				yo -= world.icon_size * (clone.transform.get_y_scale() - 1) / 2
+				xo += clone.transform.get_x_shift()
+				yo += clone.transform.get_x_shift()
 			var/icon/img = getFlatIcon(clone, no_anim = TRUE)
```

Comment:
Move the xo/yo part after getFlatIcon and add some comments what exactly these variables are please.
It's not hard to figure out but when everything is called xo, sx or rx, things can get messy.

Since photos are messy regardless, I don't really care if this makes them a bit slower but a picture as proof that it all works out properly would be nice.

Another neat thing that could be done is to add transform as an argument to getFlatIcon if you are up for that.

Reply:
Can't be done in getFlatIcon because scale affects shifting, and shifting can only be done by changing the origins, which is not a property of the icon